creating WI crosswalk w/ full hydrolayer, and polygon depth-linking for SWIMMS

jordansread opened the pull request.

Commit:
updated w/ big shapefile for hydrolayer

Commit:
adding dependencies

Commit:
local file and clean up

Commit:
update to crosswalk mapping

Commit:
running mapping w/ poly

lawinslow commented on remake_depth.yml:
```diff
@@ -40,19 +40,21 @@ targets:
     
   data/depth_data_linked/depth_swimms_wisconsin.csv:
-    command: depth_link_swimms("data/depth_data_raw/wisconsin_swims_lakes_depth.tsv")
+    command: depth_link_swimms("data/depth_data_raw/wisconsin_swims_lakes_depth.tsv", "data/NHD_state_crosswalk/nhd2wbic.RData")
```
Is there a remake that makes this new file?

lawinslow replied:
I mean, should it be in the remake now that its part of the overarching build process?

jordansread replied:
I think it is handled in the same way as in `depth_link_jacobson`, which is a load of some crosswalk file that is created by some other script that isn't captured in a remake file (e.g., https://github.com/USGS-R/necsc-lake-modeling/blob/daaaa662050bccb006d81dcc45a65c9c1466a4b6/scripts/crosswalk_MN_nhd_dowlknum.R and https://github.com/USGS-R/necsc-lake-modeling/blob/daaaa662050bccb006d81dcc45a65c9c1466a4b6/scripts/depth_link_jacobson.R#L6). I did the pattern here to add that file as a dependency so this would be re-run if the crosswalk file changes. I commented out adding the same dependencies to those other crosswalk links above, but those should probably be uncommented. 

I want to get that hydrolayer crosswalk into a remake (hence my comments about the local file on another issue), but just isn't a priority right now.

jordansread replied:
oh yeah, definitely. 

With these remake files I didn't yet know how to link to other remake targets from other files, but that really simplifies this kind of pattern... (e.g., https://github.com/jread-usgs/stream_metab_usa/blob/ec73e4578070b80405ff00fa1230f2f9898b1849/remake/4_data_release.yml#L4) 

Just a bit of work to clean these up and get them ready for prime time. Worth it?

jordansread replied:
@lawinslow created an issue for it https://github.com/USGS-R/necsc-lake-modeling/issues/95

Commit:
update with depth links

Commit:
update crosswalk

Commit:
merge conflict

Merge branch 'master' of github.com:jread-usgs/necsc-lake-modeling

# Conflicts:
#	scripts/depth_link_swimms.R

Commit:
metadata updates

Commit:
update after feedback

jordansread commented:
@lawinslow I have kind of a smashed together set of PRs that include the changes to the data release and that crosswalk. 

Your call - want me to go back and split them, or merge this to keep canonical up to date but missing the crosswalk in the make

lawinslow commented:
Let's just merge. Keep up to date.
